extendr-api/src/conversions/try_into_int.rs

```diff
@@ -0,0 +1,148 @@
+use std::fmt::{Debug, Display, Formatter};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ConversionError {
+    Underflow,
+    Overflow,
+    NotIntegerish,
+}
+
+impl Display for ConversionError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ConversionError::Underflow => write!(f, "underflow"),
+            ConversionError::Overflow => write!(f, "overflow"),
+            ConversionError::NotIntegerish => write!(f, "not a whole number"),
+        }
+    }
+}
+
+pub(crate) trait FloatToInt<T: Sized> {
+    fn try_into_int(&self) -> Result<T, ConversionError>
+    where
+        Self: Sized;
+}
+
+macro_rules! impl_into_integerish {
+    ($float_type:ty, $int_type:ty) => {
+        impl FloatToInt<$int_type> for $float_type {
+            fn try_into_int(&self) -> Result<$int_type, ConversionError> {
+                if *self == <$float_type>::default() {
```

Comment:
Might be an idea to use is zero function or something like that.

Reply:
I believe this is the easiest way for us to check for exact `0.0` and return `0` afterward. Without involving external traits.

Reply:
Is_zero doesn't exists. Lol. Probably num-traits thing